src/main/java/edu/ksu/canvas/attendance/controller/SummaryController.java

```diff
@@ -100,42 +105,50 @@ private ModelAndView studentSummary(String sectionId, String studentId, boolean
         List<AttendanceSummaryModel> summaryForSections = isSimpleAttendance ?
                 reportService.getSimpleAttendanceSummaryReport(validatedSectionId) :
                 reportService.getAviationAttendanceSummaryReport(validatedSectionId);
-        List<LtiLaunchData.InstitutionRole> institutionRoles = canvasService.getRoles();
 
         student.getAttendances().sort(Comparator.comparing(Attendance::getDateOfClass).reversed());
 
-        summaryForSections.stream()
-                .flatMap(summary -> summary.getEntries().stream())
-                .filter(entry -> entry.getStudentId() == validatedStudentId)
-                .findFirst()
-                .ifPresent(entry ->  addAssignmentSummaryToPage(page, entry, isSimpleAttendance, student, assignment));
+        List<AttendanceSummaryModel.Entry> entries = new ArrayList<>();
+        if (student.getSisUserId() != null) {
+            for (AttendanceSummaryModel model : summaryForSections) {
+                for (AttendanceSummaryModel.Entry entry : model.getEntries()) {
+                    if (student.getSisUserId().equals(entry.getSisUserId())) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS134)